src/Query/TaskGroups.ts

```diff
@@ -1,43 +1,55 @@
 import type { Task } from '../Task';
 import type { Grouper } from './Grouper';
-import { GroupHeadings } from './GroupHeadings';
-import { IntermediateTaskGroups } from './IntermediateTaskGroups';
+import { GroupDisplayHeadingSelector } from './GroupDisplayHeadingSelector';
+import { TaskGroupingTree } from './TaskGroupingTree';
 import { TaskGroup } from './TaskGroup';
 
 /**
- * TaskGroup stores all the groups of tasks generated by any 'group by'
+ * Store all the groups of tasks generated by any 'group by'
  * instructions in the task block.
+ *
+ * The groups are stored in array of {@link TaskGroup} objects.
+ * @see {@link Grouper}
+ * @see {@link Query.grouping}
  */
 export class TaskGroups {
     private _groups: TaskGroup[] = new Array<TaskGroup>();
     private _totalTaskCount = 0;
 
     /**
-     * Constructor for TaskGroups
-     * @param {Grouper[]} groups - 0 or more Grouping values,
+     * Constructor for {@link TaskGroups}
+     * @param {Grouper[]} groups - 0 or more {@link Grouper} objects,
      *                              1 per 'group by' line in the task query block
-     * @param {Task[]} tasks - 0 more more Task objects, with all the tasks
+     * @param {Task[]} tasks - 0 or more Task objects, with all the tasks
      *                         matching the query, already in sort order
      */
     constructor(groups: Grouper[], tasks: Task[]) {
         // Grouping doesn't change the number of tasks, and all the tasks
         // will be shown in at least one group.
         this._totalTaskCount = tasks.length;
 
-        const initialGroups = new IntermediateTaskGroups(groups, tasks);
-        this.addTasks(initialGroups);
+        const taskGroupingTree = new TaskGroupingTree(groups, tasks);
+        this.addTasks(taskGroupingTree);
     }
 
     /**
      * All the tasks matching the query, grouped together, and in the order
      * that they should be displayed.
+     *
+     * If the tasks are ungrouped, a single {@link TaskGroup} will be returned,
+     * with no heading names.
```

Comment:
just curious why would we need a getter here instead of just accessing the `TaskGroups.groups` And then the comment on lines 36-37 shall go the `TaskGroups.groups` itself.